src/identity/identity.ts

```diff
@@ -1546,7 +1551,22 @@ export class Identity<T extends StorageProvider> {
             // in local storage before attempting to authorize, so we remove
             // their data.
             const currentUser = await this.#getCurrentUser();
-            if (
+            const showSkipAndNoMoney =
+              this.#showSkip &&
+              e.message.indexOf('RuleErrorInsufficientBalance') >= 0;
+            if (showSkipAndNoMoney && currentUser != null) {
+              await this.#updateUser(
+                currentUser.publicKey ||
+                  currentUser.primaryDerivedKey?.publicKeyBase58Check,
```

Comment:
these are the same and I don't think you can have 1 without the other. You could just use 1 of them without the fallthrough `||` thing. 
```suggestion
                currentUser.publicKey,
```

Reply:
that's what I thought too, but there was an edge case I hit where `currentUser.publicKey` was undefined.